infra-as-code/bicep/machinelearning.bicep

```diff
@@ -326,112 +177,167 @@ resource machineLearning 'Microsoft.MachineLearningServices/workspaces@2023-10-0
+    description: 'Project to contain the "Chat with Wikipedia" example prompt flow that is used as part of the Microsoft Learn Azure OpenAI baseline chat implementation. https://learn.microsoft.com/azure/architecture/ai-ml/architecture/baseline-openai-e2e-chat'
+    v1LegacyMode: false
+    publicNetworkAccess: 'Disabled'
+    allowPublicAccessWhenBehindVnet: false
+    enableDataIsolation: true
+    hubResourceId: aiHub.id
+    primaryUserAssignedIdentity: null
+  }
+
+  resource endpoint 'onlineEndpoints' = {
+    name: 'ept-chat-${baseName}'
     location: location
+    kind: 'Managed'
     identity: {
-      type: 'UserAssigned'
-      userAssignedIdentities: {
-        '${azureMachineLearningInstanceComputeManagedIdentity.id}': {}
-      }
+      type: 'SystemAssigned' // This resource's identity is automatically assigned AcrPull access to ACR, Storage Blob Data Contributor, and AML Metrics Writer on the project. It is also assigned two additional permissions below.
+                            // Given the permissions assigned to the identity, it is recommended only include deployments in the Azure OpenAI service that are trusted to be invoked from this endpoint.
     }
     properties: {
-      computeType: 'ComputeInstance'
-      computeLocation: location
-      description: 'Machine Learning compute instance'
-      disableLocalAuth: true
-      properties: {
-        customServices: null
-        enableNodePublicIp: false
-        personalComputeInstanceSettings: null
-        schedules: {
-          computeStartStop: []
-        }
-        setupScripts: null
-        applicationSharingPolicy: 'Personal'
-        computeInstanceAuthorizationType: 'personal'
-        sshSettings: {
-          sshPublicAccess: 'Disabled'
-        }
-        vmSize: 'STANDARD_DS3_V2'
-        // idleTimeBeforeShutdown: 'PT1D' -- Some organizations use the 'Azure Machine Learning Compute Instance should have idle shutdown.' which depends on a specific preview API version not used here.
-      }
+      description: 'This is the /score endpoint for the "Chat with Wikipedia" example prompt flow deployment. Called by the UI hosted in Web Apps.'
+      authMode: 'Key' // Ideally this should be based on Microsoft Entra ID access. This sample however uses a key stored in Key Vault.
+      publicNetworkAccess: 'Disabled'
     }
-    dependsOn: [
-      // Role requirements for compute instance: https://learn.microsoft.com/azure/machine-learning/how-to-identity-based-service-authentication#pull-docker-base-image-to-machine-learning-compute-cluster-for-training-as-is
-      computeInstanceContainerRegistryPullRoleAssignment
-      computeInstanceBlobDataReaderRoleAssignment
-    ]
+
+    // Note: If you reapply this Bicep after an AI Studio managed compute deployment has happened in this endpoint, the taffic routing reverts to 0% to all existing deployments. You'll need to set that back to 100% to your desired deployment.
   }

```

Comment:
Can you add a "logical" depends on here for `aiHub::aoaiConnection`.  That helps make sure that the "hub" is established fully before we get into deploying any projects.  It's not necessary, but adds a bit of logical ordering to the deployment.  This will be something I backport to Baseline and Basic then as well.

Reply:
Let's defer this for now if not absolutely necessary, don't want to introduce a change above and beyond what has been tested end to end and is ready for this release